File: 10_15_2016_Search-in-Rotated-Sorted-Array/chang.java
public class Solution {
    /** 
     *@param A : an integer rotated sorted array
     *@param target :  an integer to be searched
     *return : an integer
     */
    public int search(int[] A, int target) {
        // write your code here
        if (A == null || A.length == 0) {
            return -1;
        }
        
        int start = 0;
        int end = A.length -1;
        
        while (start + 1 < end) {
            int mid = start + (end - start) / 2;
            if (A[mid] == target) {
                return mid;
            }
            if (A[start] < A[mid]) {
                //start...target...mid
                if (A[start] <= target && A[mid] >= target) {
                    end = mid;
                }else {
                    start = mid;
                }
            }else {//A[start] >= A[mid]
                //mid...target...end
                if (A[mid] <= target && A[end] >= target) {
                    start = mid;
                }else {
                    end = mid;
                }
                
            }
        }
        if (A[start] == target) {
            return start;
        }
        if (A[end] == target) {
            return end;
        }
        return -1;
    }
    
}

File: 01_02_2017_reverse_integer/chang.py
class Solution:
    # @param {int} n the integer to be reversed
    # @return {int} the reversed integer
    def reverseInteger(self, n):
        if n is None or n == 0:
            return 0
            
        if n < 0:
            negative = -1
            n = -n
        else:
            negative = 1
        
        reversed = 0
        
        while n > 0:
            reversed = reversed * 10 + n % 10
            n = n / 10
        
        reversed = reversed * negative
        
        if reversed < -(1 << 31) or reversed > (1 << 31) - 1:
            return 0
        
        return reversed 
